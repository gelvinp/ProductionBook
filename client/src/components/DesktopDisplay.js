import React, { Component } from 'react'
import { Grid, Segment, Button } from 'semantic-ui-react'
import PDFViewer from '../containers/PDFViewerContainer.js'
import DocumentColumn from '../containers/DocumentColumnContainer.js'
import PropTypes from 'prop-types'

class DesktopDisplay extends Component {
  render() {
    const { logOut } = this.props
    return (
      <Grid
        style={{ flexDirection: 'column', height: '100%', margin: 'unset' }}
      >
        <Grid.Row style={{ flexGrow: 1 }}>
          <Grid.Column width={4} style={{ height: '100%' }}>
            <DocumentColumn />
          </Grid.Column>
          <Grid.Column width={12}>
            <PDFViewer />
          </Grid.Column>
        </Grid.Row>
        <Grid.Row style={{ padding: 0 }}>
          <Grid.Column style={{ padding: 0 }}>
            <Segment inverted attached style={{ padding: '0.5em' }}>
              <div
                style={{
                  display: 'flex',
                  flexDirection: 'row',
                  justifyContent: 'space-between',
                  alignItems: 'center',
                }}
              >
                <p style={{ marginBottom: 0 }}>
                  {process.env.REACT_APP_PROJECT} Production Book
                </p>
                <Button floated="right" inverted basic onClick={logOut}>
                  Log Out
                </Button>
              </div>
            </Segment>
          </Grid.Column>
        </Grid.Row>
      </Grid>
    )
  }
}

DesktopDisplay.propTypes = {
  logOut: PropTypes.func.isRequired,
}

export default DesktopDisplay
