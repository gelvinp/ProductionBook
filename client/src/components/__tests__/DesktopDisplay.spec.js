import React from 'react'
import DesktopDisplay from '../DesktopDisplay.js'
import { shallow } from 'enzyme'

describe('DocumentColumn', () => {
  it('Displays correctly', () => {
    const wrapper = shallow(<DesktopDisplay />)
    expect(wrapper).toMatchSnapshot()
  })
})
