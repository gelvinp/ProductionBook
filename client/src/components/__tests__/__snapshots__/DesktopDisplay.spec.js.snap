// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentColumn Displays correctly 1`] = `
<Grid
  style={
    Object {
      "flexDirection": "column",
      "height": "100%",
      "margin": "unset",
    }
  }
>
  <GridRow
    style={
      Object {
        "flexGrow": 1,
      }
    }
  >
    <GridColumn
      style={
        Object {
          "height": "100%",
        }
      }
      width={4}
    >
      <Connect(DocumentColumn) />
    </GridColumn>
    <GridColumn
      width={12}
    >
      <Connect(PDFViewer) />
    </GridColumn>
  </GridRow>
  <GridRow
    style={
      Object {
        "padding": 0,
      }
    }
  >
    <GridColumn
      style={
        Object {
          "padding": 0,
        }
      }
    >
      <Segment
        attached={true}
        inverted={true}
        style={
          Object {
            "padding": "0.5em",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <p
            style={
              Object {
                "marginBottom": 0,
              }
            }
          >
            Development
             Production Book
          </p>
          <Button
            as="button"
            basic={true}
            floated="right"
            inverted={true}
            onClick={[MockFunction]}
          >
            Log Out
          </Button>
        </div>
      </Segment>
    </GridColumn>
  </GridRow>
</Grid>
`;
